c2h/include/c2h/bfloat16.cuh

```diff
@@ -232,19 +233,36 @@ inline std::ostream& operator<<(std::ostream& out, const __nv_bfloat16& x)
  * Traits overloads
  ******************************************************************************/
 
+_LIBCUDACXX_BEGIN_NAMESPACE_STD
 template <>
-struct CUB_NS_QUALIFIER::FpLimits<bfloat16_t>
+struct __is_extended_floating_point<bfloat16_t> : true_type
```

Comment:
question: Can you help me understand where this is used? My mental model of of our custom `bfloat16_t` and `half_t` is that we use them to "emulate" the native extended fp types in the absence of those. The reason I am asking is that I would like to make sure we're not promoting these "emulated" wrapper types to be a "real extended fp type" in places where it isn't.

Reply:
It's used by `cuda::is_floating_point` and `cuda::std::numeric_limits`. Adding `bfloat16_t` here allows `__numeric_limits_impl` to be chosen internally by `cuda::std::numeric_limits`. It also makes sure that since `cuda::is_floating_point<__nv_bfloat16>` is true, `cuda::is_floating_point<bfloat16_t>` is also true. So in short, it alignes the traits and limits for `__nv_bfloat16` and its wrapper type.

The same goes for `__half` and `half_t`.